airbyte-webapp/src/core/form/schemaToFormBlock.ts

```diff
@@ -37,20 +38,49 @@ export const jsonSchemaToUiWidget = (
   }
 
   if (jsonSchema.oneOf?.length && jsonSchema.oneOf.length > 0) {
-    const conditions = Object.fromEntries(
-      jsonSchema.oneOf.map((condition) => {
-        if (typeof condition === "boolean") {
-          return [];
-        }
-        return [condition.title, jsonSchemaToUiWidget({ ...condition, type: jsonSchema.type }, key, path)];
-      })
+    let possibleConditionSelectionKeys = null as Set<string> | null;
+    const conditions = jsonSchema.oneOf.flatMap((condition) => {
+      if (typeof condition === "boolean") {
+        throw new FormBuildError("Spec uses oneOf without using object types for all conditions");
+      }
+      const formBlock = jsonSchemaToFormBlock({ ...condition, type: jsonSchema.type }, key, path);
+      if (formBlock._type !== "formGroup") {
+        return [];
+      }
+
+      const constProperties = new Set(
+        formBlock.properties.filter((property) => property.const).map((property) => property.fieldKey)
+      );
```

Comment:
The `Set` class is an ordered set in Javascript: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

I know it's not important for the amount of data we are dealing with, but theoretically the intersect is slightly faster this way because the `has` method is constant time now instead of linear time for ``, reducing the algorithm from `n*n` to `n`.

That said, it's probably better to stick with arrays here to keep the logic simple as there's a lot of stuff going on already - rewrote it using arrays and the lodash intersection helper which I hope will do if fast (if not its easy to refactor later on)